Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py

```diff
@@ -1110,70 +1156,137 @@ def fetch_incidents():
 
-    # Handle first time fetch, fetch incidents retroactively
-    if last_fetch is None:
-        last_fetch, _ = parse_date_range(FETCH_TIME, date_format='%Y-%m-%dT%H:%M:%SZ')
+        last_run = demisto.getLastRun()
+        # Get the last fetch time, if exists
+        last_fetch = last_run.get('first_behavior_time')
 
-    last_fetch_timestamp = int(parse(last_fetch).timestamp() * 1000)
+        # Handle first time fetch, fetch incidents retroactively
+        if last_fetch is None:
+            last_fetch, _ = parse_date_range(FETCH_TIME, date_format='%Y-%m-%dT%H:%M:%SZ')
 
-    last_detection_id = str(last_run.get('last_detection_id'))
+        last_fetch_timestamp = int(parse(last_fetch).timestamp() * 1000)
 
-    fetch_query = demisto.params().get('fetch_query')
+        last_detection_id = str(last_run.get('last_detection_id'))
 
-    if fetch_query:
-        fetch_query = "created_timestamp:>'{time}'+{query}".format(time=last_fetch, query=fetch_query)
-        detections_ids = demisto.get(get_fetch_detections(filter_arg=fetch_query), 'resources')
+        fetch_query = demisto.params().get('fetch_query')
 
-    else:
-        detections_ids = demisto.get(get_fetch_detections(last_created_timestamp=last_fetch), 'resources')
-    incidents = []  # type:List
+        if fetch_query:
+            fetch_query = "created_timestamp:>'{time}'+{query}".format(time=last_fetch, query=fetch_query)
+            detections_ids = demisto.get(get_fetch_detections(filter_arg=fetch_query), 'resources')
 
-    if detections_ids:
+        else:
+            detections_ids = demisto.get(get_fetch_detections(last_created_timestamp=last_fetch), 'resources')
 
-        # make sure we do not fetch the same detection again.
-        if last_detection_id == detections_ids[0]:
-            first_index_to_fetch = 1
+        if detections_ids:
+
+            # make sure we do not fetch the same detection again.
+            if last_detection_id == detections_ids[0]:
+                first_index_to_fetch = 1
+
+                # if this is the only detection - dont fetch.
+                if len(detections_ids) == 1:
+                    return incidents
+
+            # if the first detection in this pull is different than the last detection fetched we bring it as well
+            else:
+                first_index_to_fetch = 0
 
-            # if this is the only detection - dont fetch.
-            if len(detections_ids) == 1:
-                return incidents
+            # Limit the results to INCIDENTS_PER_FETCH`z
+            last_index_to_fetch = INCIDENTS_PER_FETCH + first_index_to_fetch
+            detections_ids = detections_ids[first_index_to_fetch:last_index_to_fetch]
+            raw_res = get_detections_entities(detections_ids)
+
+            if "resources" in raw_res:
+                raw_res['type'] = "detections"
+                for detection in raw_res.get("resources"):
```

Comment:
why did you change these lines?

Reply:
The first line - we decided to add to each incident its type - incident or detection.
Second line - my mistake